File: Projects/EspressoDemos/EspressoEffectiveTesting/app/src/androidTest/java/com/example/code/injection/TestRecipeApplication.java
package com.example.code.injection;

import com.example.code.local.Favorites;
import com.example.code.local.InMemoryFavorites;

public class TestRecipeApplication extends RecipeApplication{
    private final Favorites favorites = new InMemoryFavorites();
    @Override
    public Favorites getFavorites() {
        return favorites;
    }
}

File: Projects/EspressoDemos/EspressoEffectiveTesting/app/src/androidTest/java/com/example/code/injection/TestRecipeApplication.kt
package com.example.code.injection

import com.example.code.local.Favorites
import com.example.code.local.InMemoryFavorites

class TestRecipeApplication : RecipeApplication() {
    private val favorites: Favorites = InMemoryFavorites()
    override fun getFavorites(): Favorites {
        return favorites
    }
}